core/computed/entity-classification.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {makeComputedArtifact} from './computed-artifact.js';
+import {NetworkRecords} from './network-records.js';
+import {Util} from '../util.cjs';
+import UrlUtils from '../lib/url-utils.js';
+import thirdPartyWeb from '../lib/third-party-web.js';
+
+/** @typedef {Map<string, LH.Artifacts.RecognizableEntity>} EntityCache */
+
+class EntityClassification {
+  /**
+   * @param {EntityCache} entityCache
+   * @param {string} url
+   * @return Entity | undefined
+   */
+  static makeUpAnEntity(entityCache, url) {
+    if (!UrlUtils.isValid(url)) return;
+    const rootDomain = Util.getRootDomain(url);
+    if (!rootDomain) return;
```

Comment:
In what cases would `Util.getRootDomain` return undefined?

Reply:
The only case I can think of `L22` returning `true` and rootDomain going empty string is `data:` URIs.

Reply:
Now, looking at code of `Util.getRootDomain` — `chrome://`, `file://` et al.,  could make their way here and `Util.getRootDomain` would return non falsey output — but they're not entities-worthy either. Seems like an issue with `getRootDomain`, I think? Or should we limit protocols here? wdyt?